src/kernels/jupyter/jupyterDetectionTelemetry.node.ts

```diff
@@ -0,0 +1,117 @@
+        this.detectJupyter('lab', process.env).catch(noop);
+        const { env, shell } = await this.terminalHelper.getEnvironmentVariables(undefined);
+        if (!env) {
+            return;
+        }
+        const mergedVariables = { ...process.env };
+        Object.keys(env).forEach((key) => {
+            if (key in process.env) {
+                return;
+            }
+            if (key.toLowerCase() == 'path') {
+                // Append path from terminal.
+                let delimiter = path.delimiter;
+                const currentPath = (process.env['PATH'] || process.env['Path'] || '').trim();
+                const terminalPath = (env['PATH'] || env['Path'] || '').trim();
+                if (currentPath.endsWith(delimiter)) {
+                    delimiter = '';
+                }
+                if (terminalPath.startsWith(delimiter)) {
+                    delimiter = '';
+                }
+                mergedVariables[key] = `${currentPath}${delimiter}${terminalPath}`;
+            } else {
+                mergedVariables[key] = env[key];
+            }
+        });
+        this.detectJupyter('notebook', mergedVariables, shell).catch(noop);
+        this.detectJupyter('lab', mergedVariables, shell).catch(noop);
+    }
+    private async detectJupyter(
+        frontEnd: 'notebook' | 'lab',
+        env: NodeJS.ProcessEnv,
+        shell?: TerminalShellType
+    ): Promise<void> {
+        try {
+            const processService = await this.processFactory.create(undefined);
+            const output = await processService.exec('jupyter', [frontEnd, '--version'], {
+                env,
+                throwOnStdErr: false,
+                mergeStdOutErr: true
+            });
+            const versionLines = output.stdout
+                .splitLines({ trim: true, removeEmptyEntries: true })
+                .filter((line) => !isNaN(parseInt(line.substring(0, 1), 10)));
+            const versionMatch = /^\s*(\d+)\.(\d+)\.(.+)\s*$/.exec(versionLines.length ? versionLines[0] : '');
+            if (versionMatch && versionMatch.length > 2) {
+                const major = parseInt(versionMatch[1], 10);
+                const minor = parseInt(versionMatch[2], 10);
+                const frontEndVersion = parseFloat(`${major}.${minor}`);
+                if (shell) {
+                    sendTelemetryEvent(Telemetry.JupyterInstalled, undefined, {
+                        frontEnd,
+                        frontEndVersion,
+                        detection: 'shell',
+                        shellType: shell
+                    });
+                } else {
+                    sendTelemetryEvent(Telemetry.JupyterInstalled, undefined, {
+                        frontEnd,
+                        frontEndVersion,
+                        detection: 'process'
+                    });
+                }
+            } else {
+                sendTelemetryEvent(Telemetry.JupyterInstalled, undefined, {
```

Comment:
Is it interesting if it's installed but installed incorrectly? Maybe we would detect that elsewhere. 

I'm thinking of the two users that had juptyer installed by their kernels were invalid we had during endgame week.

Reply:
> Is it interesting if it's installed but installed incorrectly? Maybe we would detect that elsewhere

Please could you share the link to that issue.